shared/deps/serial/VirtIoChannel_guest_send_receive.py

```diff
@@ -86,22 +86,23 @@ def shake_hand(self, size=0, action="receive"):
             txt = self.receive(hi_msg_len)
             out = struct.unpack(self.hi_format, txt)[0]
             if out != "HI":
-                raise ShakeHandError("Fail to get HI from guest.")
+                raise ShakeHandError("Fail to get HI from host.")
             size_s = struct.pack("q", size)
             self.send(size_s)
             txt = self.receive(ack_msg_len)
             ack_str = struct.unpack(self.ack_format, txt)[0]
             if ack_str != self.ack_msg:
-                raise ShakeHandError("Guest didn't ACK the file size message.")
+                raise ShakeHandError("Host didn't ACK the file size message.")
             return size
         elif action == "receive":
-            txt = self.receive(hi_msg_len)
+            txt = ''
+            while len(txt) < hi_msg_len:
```

Comment:
why do this update ，   txt = self.receive(hi_msg_len) looks OK in here？

Reply:
@xutian If start guest script before host one, later line `hi_str = struct.unpack(self.hi_format, txt)[0]`  will raise error since `txt` will be None as nothing received.

Reply:
Ok, better to comments here.

Reply:
@xutian Explained it in commit message: https://github.com/avocado-framework/avocado-vt/pull/2135/commits/5eaf9039bc7d6bd410fe025b425c5d1d38c16e6b